# An Introduction to Pandas Part II
### Harsha Kankanamge

In [1]:
import pandas as pd
pd.__version__

'0.23.4'

In [2]:
 # get working (current) directory
import os
os.getcwd()

'C:\\Users\\harsh\\Desktop\\GEG584\\Assignment_3'

In [3]:
os.chdir('data') # change directory to where data file is stored
os.getcwd()

'C:\\Users\\harsh\\Desktop\\GEG584\\Assignment_3\\data'

In [4]:
filename = 'PlacePop_poverty2017.csv'
pop_df = pd.read_csv(filename,encoding='UTF-8')
pop_df.shape # How many rows and columns in the dataframe

(29323, 7)

In [5]:
pop_df.head()

Id     Id2                 Geography  PopEst  PopMoe  \
0  1600000US0100100  100100       Abanda CDP, Alabama     174     158   
1  1600000US0100124  100124   Abbeville city, Alabama    2412      81   
2  1600000US0100460  100460  Adamsville city, Alabama    4379      31   
3  1600000US0100484  100484     Addison town, Alabama     720     198   
4  1600000US0100676  100676       Akron town, Alabama     318     125   

   Under18Est  Under18Moe  
0          43          67  
1         499         118  
2         950         220  
3         141          77  
4          58          38

In [6]:
pop_df.columns

Index(['Id', 'Id2', 'Geography', 'PopEst', 'PopMoe', 'Under18Est',
       'Under18Moe'],
      dtype='object')

In [7]:
fields=[]
for x in [2,5,6]:fields.append(pop_df.columns[x])
fields

['Geography', 'Under18Est', 'Under18Moe']

In [8]:
pop_df = pd.read_csv(filename,usecols=fields)
pop_df.shape

(29323, 3)

### Find places similar in size to Buffalo, NY.
Calculate the z-score comparing the under 18 population of all places to the under 18 population of Buffalo, NY.                
Use this information to retain places with a under 18 population that is not significantly different from Buffalo's.                   

1.Get Buffalo's population (Under18Est and Under18Moe)                                
2.Calculate z-score comparing Buffalo to all other places

In [9]:
bufEst = pop_df.loc[pop_df.Geography=='Buffalo city, New York',['Under18Est']].values[0,0]
bufMoe = pop_df.loc[pop_df.Geography=='Buffalo city, New York',['Under18Moe']].values[0,0]
bufEst,bufMoe

(58617, 1157)

In [10]:
pop_df['zBflo']=(pop_df['Under18Est']-bufEst)/((pop_df['Under18Moe']/1.645)**2+(bufMoe/1.645)**2)**.5
pop_df['zBflo'][:10]

0   -83.140086
1   -82.204616
2   -80.546627
3   -82.956526
4   -83.213173
5   -65.144343
6   -68.131156
7   -75.606464
8   -80.195862
9   -81.682201
Name: zBflo, dtype: float64

In [11]:
pop_df.loc[(pop_df['zBflo'].abs() < 1.645)] # under 18 population same as Buffalo, 90% confidence

Geography  Under18Est  Under18Moe     zBflo
2405         Fontana city, California       59811        1396  1.083277
2839   Moreno Valley city, California       58287        1175 -0.329195
5798       Urban Honolulu CDP, Hawaii       59530        1069  0.953427
6110            Aurora city, Illinois       58957        1237  0.330212
15667          Henderson city, Nevada       59820        1404  1.087743
16959          Buffalo city, New York       58617        1157  0.000000
24703         Brownsville city, Texas       57089        1119 -1.561606

## Result
Only seven places have Under 18 population equal to Buffalo's (not surprising).                     

Define medium size places as those with under 18 populations between 30,000 and 80,000.                                   

Find all places with populations >= 30,000 but less than 80,000.                                   

In [12]:
pop_df.loc[pop_df.Geography=='Buffalo city, New York']

Geography  Under18Est  Under18Moe  zBflo
16959  Buffalo city, New York       58617        1157    0.0

In [13]:
pop_df['zLB']=(pop_df['Under18Est']-30000)/(pop_df['Under18Moe']/1.645)
pop_df['zUB']=(pop_df['Under18Est']-80000)/(pop_df['Under18Moe']/1.645)
pop_df[:10]

Geography  Under18Est  Under18Moe      zBflo  \
0           Abanda CDP, Alabama          43          67 -83.140086   
1       Abbeville city, Alabama         499         118 -82.204616   
2      Adamsville city, Alabama         950         220 -80.546627   
3         Addison town, Alabama         141          77 -82.956526   
4           Akron town, Alabama          58          38 -83.213173   
5       Alabaster city, Alabama        8687         501 -65.144343   
6     Albertville city, Alabama        6447         498 -68.131156   
7  Alexander City city, Alabama        3564         310 -75.606464   
8       Alexandria CDP, Alabama         837         257 -80.195862   
9      Aliceville city, Alabama         535         172 -81.682201   

           zLB          zUB  
0  -735.511418 -1963.123358  
1  -411.263941 -1108.297839  
2  -217.214773  -591.078409  
3  -637.896818 -1706.078636  
4 -1296.173421 -3460.647105  
5   -69.979810  -234.151467  
6   -77.800572  -242.961215  
7  -140.281355  -405.603935  
8  -186.665895  -506.704805  
9  -281.801890  -759.999564

In [14]:
lbdf=pop_df.loc[(pop_df.zLB >= -1.645)]
medPlaces = lbdf.loc[(lbdf.zUB < -1.645)].copy() # population within (30000, 80000), 90% confidence
medPlaces.reset_index(inplace=True)
medPlaces = medPlaces[fields]
medPlaces.tail(10)

Geography  Under18Est  Under18Moe
151      Arlington CDP, Virginia       39441         213
152    Chesapeake city, Virginia       57010         202
153  Newport News city, Virginia       41268         264
154       Norfolk city, Virginia       48474         185
155      Richmond city, Virginia       38716         341
156        Kent city, Washington       30344        1038
157     Spokane city, Washington       45733         913
158      Tacoma city, Washington       43458        1119
159   Vancouver city, Washington       39000         973
160      Madison city, Wisconsin       40824         787

### Result
This new dataframe contains under 18 population estimates for 161 places similar in size to Buffalo. We are interested in how Buffalo compares to these cities with respect to the percent of the under 16 population below the poverty level.                         

Since we only need the data for these 161 places, join the poverty rate data to this dataframe.

Analyze the results for these places.

In [15]:
filename = 'poverty2017_5yr.csv'
fields = ['Geography', 'Under 18 BelPov Est','Under 18 BelPov MOE' ]
df = pd.read_csv(filename,usecols=fields)
df.columns=["Geography", "est","moe"]
df.columns

Index(['Geography', 'est', 'moe'], dtype='object')

Since the population dataframe and the poverty rate dataframe have a common field (Geography), a simple merge statement suffices.

In [16]:
mergedf = pd.merge(medPlaces,df)
mergedf.tail()

Geography  Under18Est  Under18Moe   est  moe
156       Kent city, Washington       30344        1038  21.2  2.8
157    Spokane city, Washington       45733         913  24.1  2.5
158     Tacoma city, Washington       43458        1119  21.7    2
159  Vancouver city, Washington       39000         973  20.5  2.5
160     Madison city, Wisconsin       40824         787  16.5  2.4

In [17]:
mergedf['est']=mergedf['est'].apply(lambda x: pd.to_numeric(x, errors='coerce'))
mergedf['moe']=mergedf['moe'].apply(lambda x: pd.to_numeric(x, errors='coerce'))

In [18]:
mergedf.dropna(inplace=True)
mergedf[:5]

Geography  Under18Est  Under18Moe   est  moe
0        Birmingham city, Alabama       42202        1287  45.4  2.9
1        Huntsville city, Alabama       39934        1048  28.7  2.3
2            Mobile city, Alabama       41863        1086  34.8  2.7
3        Montgomery city, Alabama       47933         448  33.5  2.4
4  Anchorage municipality, Alaska       72324         363  11.5  1.5

## Calculate z-score

In [19]:
bufEst = mergedf.loc[mergedf.Geography=='Buffalo city, New York',['est']].values[0,0]
bufMoe = mergedf.loc[mergedf.Geography=='Buffalo city, New York',['moe']].values[0,0]
bufEst,bufMoe

(47.2, 2.4)

In [20]:
mergedf['zBflo2']=(mergedf['est']-bufEst)/((mergedf['moe']/1.645)**2+(bufMoe/1.645)**2)**.5
mergedf['zBflo2'][:10]

0    -0.786599
1    -9.154961
2    -5.646539
3    -6.639880
4   -20.749989
5   -20.131630
6   -25.940385
7    -9.109344
8   -19.829989
9   -16.186344
Name: zBflo2, dtype: float64

In [21]:
mergedf.loc[mergedf.zBflo2 > 1.645,['Geography','est','moe']] # percent poverty greater than Buffalo, 90% confidence

Geography   est  moe
108  Rochester city, New York  51.9  2.3

### Result
Only 1 Place in the U.S. similar in size to Buffalo has a higher proportion of population below the poverty level.

How many similarly sized places have a poverty rate that is not significantly different from Buffalo's?

In [22]:
mergedf['zBflo2']=mergedf['zBflo2'].abs()

In [23]:
mergedf.loc[mergedf.zBflo2 <= 1.645,['Geography','est','moe']] # percent poverty similar to Buffalo, 90% confidence

Geography   est  moe
0    Birmingham city, Alabama  45.4  2.9
107    Buffalo city, New York  47.2  2.4
109   Syracuse city, New York  46.8  2.7
118         Dayton city, Ohio  49.4  3.2

### Result Only 3!
How many places with under 18 population greater than 80000 have higher poverty rates than Buffalo?

In [24]:
bigdf=pop_df.loc[(pop_df.zUB >= 1.645)].copy() # under 18 population greater than or equal to 80000, 90% confidence
bigdf.shape

(54, 6)

In [25]:
bigMergedf = pd.merge(bigdf,df)
bigMergedf.tail()

Geography  Under18Est  Under18Moe       zBflo  \
49             Laredo city, Texas       85469         425   35.836409   
50        San Antonio city, Texas      366098        1669  249.065630   
51  Virginia Beach city, Virginia      100600         244   58.405945   
52       Seattle city, Washington      103576        1479   39.385453   
53      Milwaukee city, Wisconsin      154525        1026  102.023786   

           zLB         zUB   est  moe  
49  214.697659   21.168247  40.8    2  
50  331.264955  281.983948  27.2  0.9  
51  475.971311  138.881148  11.4  1.2  
52   81.834023   26.222123  13.4  1.4  
53  199.652656  119.486964  39.8  1.2

In [26]:
bigMergedf['est']=bigMergedf['est'].apply(lambda x: pd.to_numeric(x, errors='coerce'))
bigMergedf['moe']=bigMergedf['moe'].apply(lambda x: pd.to_numeric(x, errors='coerce'))

In [27]:
bigMergedf['zBflo2']=(bigMergedf['est']-bufEst)/((bigMergedf['moe']/1.645)**2+(bufMoe/1.645)**2)**.5
bigMergedf['zBflo2'][:10]

0   -14.464353
1   -10.717653
2    -8.115333
3   -12.531930
4   -12.137174
5    -4.049138
6   -11.463053
7   -11.876868
8   -11.633789
9   -11.624643
Name: zBflo2, dtype: float64

In [28]:
bigMergedf.loc[bigMergedf.zBflo2 > 1.645,['Geography','est','moe']] # percent poverty greater than Buffalo, 90% confidence

Geography   est  moe
27  Detroit city, Michigan  54.5  1.4
35    Cleveland city, Ohio  51.6  1.8

In [29]:
bigMergedf['zBflo2']=bigMergedf['zBflo2'].abs()
bigMergedf.loc[bigMergedf.zBflo2 <= 1.645,['Geography','est','moe']] # percent poverty similar to Buffalo, 90% confidence

Empty DataFrame
Columns: [Geography, est, moe]
Index: []

### Summary
Fifty four large places (Mesa city,Phoenix city, Arizona , Anaheim city, California and more) and 1 similar sized place (Rochester, NY) have under 18 (childrens) poverty rates higher than Buffalo's. Three similary sized places (Birmingham city, Alabama, Syracuse city, New York and Dayton city, Ohio) have poverty rates that are not significantly different from Buffalo's for under 18 (childrens).



# Challenge:

Instead of using total population as the measure of size for the comparisons, use the proportion of children in the population. These values can be calculated from the PlacePop_poverty2017.csv file.

In [30]:
import pandas as pd
df_2=pd.read_csv('PlacePop_poverty2017.csv')

In [31]:
df_2[:5]

Id     Id2                 Geography  PopEst  PopMoe  \
0  1600000US0100100  100100       Abanda CDP, Alabama     174     158   
1  1600000US0100124  100124   Abbeville city, Alabama    2412      81   
2  1600000US0100460  100460  Adamsville city, Alabama    4379      31   
3  1600000US0100484  100484     Addison town, Alabama     720     198   
4  1600000US0100676  100676       Akron town, Alabama     318     125   

   Under18Est  Under18Moe  
0          43          67  
1         499         118  
2         950         220  
3         141          77  
4          58          38

 ### Calculate the estimated proportion of children PctKidEst = (Under18Est/PopEst) and the margin of error

In [32]:
df_2['PctKidEst'] = (df_2['Under18Est']/df_2['PopEst']) 

### PctKidMoe = ((1/PopEst) * sqrt( Under18Moe^2 - PctKidEst^2 * PopMoe^2) 

In [33]:
df_2['PctKidMoe'] = ((1/df_2['PopEst']) * ( df_2['Under18Moe']**2 - (df_2['PctKidEst']**2 * df_2['PopMoe']**2))**0.5)

In [34]:
df_2[:5]

Id     Id2                 Geography  PopEst  PopMoe  \
0  1600000US0100100  100100       Abanda CDP, Alabama     174     158   
1  1600000US0100124  100124   Abbeville city, Alabama    2412      81   
2  1600000US0100460  100460  Adamsville city, Alabama    4379      31   
3  1600000US0100484  100484     Addison town, Alabama     720     198   
4  1600000US0100676  100676       Akron town, Alabama     318     125   

   Under18Est  Under18Moe  PctKidEst  PctKidMoe  
0          43          67   0.247126   0.312910  
1         499         118   0.206882   0.048426  
2         950         220   0.216945   0.050216  
3         141          77   0.195833   0.092395  
4          58          38   0.182390   0.095600

In [35]:
df_2['PctKidEst']=df_2['PctKidEst']*100
df_2['PctKidMoe']=df_2['PctKidMoe']*100

In [36]:
df_2=df_2[["Geography", "PctKidEst","PctKidMoe"]]
df_2[:5]

Geography  PctKidEst  PctKidMoe
0       Abanda CDP, Alabama  24.712644  31.291041
1   Abbeville city, Alabama  20.688226   4.842623
2  Adamsville city, Alabama  21.694451   5.021630
3     Addison town, Alabama  19.583333   9.239504
4       Akron town, Alabama  18.238994   9.560045

### Find places similar in size to Buffalo, NY.
Calculate the z-score comparing the under 18 population of all places to the under 18 population of Buffalo, NY.                
Use this information to retain places with a under 18 population that is not significantly different from Buffalo's.                   

1.Get Buffalo's population (Under18Est and Under18Moe)                                
2.Calculate z-score comparing Buffalo to all other places

In [37]:
bufEst = df_2.loc[df_2.Geography=='Buffalo city, New York',['PctKidEst']].values[0,0]
bufMoe = df_2.loc[df_2.Geography=='Buffalo city, New York',['PctKidMoe']].values[0,0]
bufEst,bufMoe

(23.324022346368714, 0.45658900448776835)

In [38]:
df_2['zBflo2']=(df_2['PctKidEst']-bufEst)/((df_2['PctKidMoe']/1.645)**2+(bufMoe/1.645)**2)**.5
df_2['zBflo2'][:10]

0    0.072993
1   -0.891406
2   -0.531627
3   -0.665180
4   -0.873986
5    3.735851
6    5.010986
7    1.044431
8   -0.964435
9   -0.395239
Name: zBflo2, dtype: float64

In [39]:
df_3=df_2.loc[(df_2['zBflo2'].abs() < 1.645)].copy() # percent under 18 same as Buffalo, 90% confidence

In [40]:
df_3

Geography  PctKidEst  PctKidMoe    zBflo2
0                  Abanda CDP, Alabama  24.712644  31.291041  0.072993
1              Abbeville city, Alabama  20.688226   4.842623 -0.891406
2             Adamsville city, Alabama  21.694451   5.021630 -0.531627
3                Addison town, Alabama  19.583333   9.239504 -0.665180
4                  Akron town, Alabama  18.238994   9.560045 -0.873986
7         Alexander City city, Alabama  24.712245   2.138275  1.044431
8              Alexandria CDP, Alabama  20.379839   5.000982 -0.964435
9             Aliceville city, Alabama  21.953221   5.687021 -0.395239
10               Allgood town, Alabama  21.237458  12.134203 -0.282670
11               Altoona town, Alabama  18.719212   6.598968 -1.145156
12             Andalusia city, Alabama  21.845398   2.019872 -1.174568
13              Anderson town, Alabama  15.702479   8.748035 -1.431224
15                  Arab city, Alabama  23.021583   2.035995 -0.238437
16               Ardmore town, Alabama  20.939597   7.322344 -0.534634
17                  Argo town, Alabama  21.570031   6.026889 -0.477373
18                Ariton town, Alabama  23.702665   8.018948  0.077549
20               Ashford town, Alabama  21.910367   3.547554 -0.650149
21               Ashland town, Alabama  20.051151   7.506219 -0.715932
22              Ashville city, Alabama  26.840753   6.532959  0.883359
24                Atmore city, Alabama  23.179253   4.141127 -0.057161
25               Attalla city, Alabama  24.952049   5.913156  0.451562
27          Autaugaville town, Alabama  18.050941   8.458001 -1.024073
28                  Avon town, Alabama  26.211454   7.418047  0.639097
29                   Axis CDP, Alabama  34.056886  17.985516  0.981338
30                Babbie town, Alabama  25.631769  10.294114  0.368416
31             Baileyton town, Alabama  28.784404  13.737942  0.653473
32             Bakerhill town, Alabama  15.744681  11.706505 -1.064241
33               Ballplay CDP, Alabama  22.022741   7.873558 -0.271417
34                 Banks town, Alabama  25.454545  23.088046  0.151768
37            Bear Creek town, Alabama  19.071730   7.493730 -0.931722
...                                ...        ...        ...       ...
29277          Riverside town, Wyoming   0.000000  27.906977 -1.374670
29278           Riverton city, Wyoming  22.688284   2.054700 -0.496855
29280         Rock River town, Wyoming  10.937500  13.094589 -1.555104
29282      Rolling Hills town, Wyoming  27.906977  14.899854  0.505738
29284           Saratoga town, Wyoming  22.743682   8.065234 -0.118178
29285               Shell CDP, Wyoming  27.218935  30.785285  0.208100
29286           Sheridan city, Wyoming  23.217828   1.280499 -0.128499
29287           Shoshoni town, Wyoming  17.261905   9.620123 -1.035431
29288           Sinclair town, Wyoming  22.926829  11.430810 -0.057114
29289              Slater CDP, Wyoming  32.098765  30.016165  0.480834
29290       Sleepy Hollow CDP, Wyoming  16.386555   9.348691 -1.219267
29291               Smoot CDP, Wyoming  60.231660  51.338359  1.182559
29292       South Greeley CDP, Wyoming  25.603757   4.968632  0.751601
29293          South Park CDP, Wyoming  31.602280   8.939571  1.521326
29294  Star Valley Ranch town, Wyoming  20.282069   8.412986 -0.593922
29295               Story CDP, Wyoming  18.573046  15.572586 -0.501651
29296           Sundance town, Wyoming  27.817404  10.342528  0.713986
29297           Superior town, Wyoming  29.765886  11.047647  0.958379
29301       Teton Village CDP, Wyoming  34.090909  47.326876  0.374221
29302             Thayne town, Wyoming  30.975610  11.738305  1.071479
29303        Thermopolis town, Wyoming  20.484041   3.666981 -1.264247
29304         Torrington city, Wyoming  22.733119   2.279210 -0.418171
29305         Turnerville CDP, Wyoming  16.071429  15.966534 -0.746915
29306              Upton town, Wyoming  21.151717   7.760040 -0.459698
29308        Van Tassell town, Wyoming  14.285714  16

## Result
18804 places have percent population under 18 equal to Buffalo'.   
Comapre to poverty rate for under 18 in Buffalo to the rate for these places

In [41]:
filename = 'poverty2017_5yr.csv'
fields = ['Geography', 'Under 18 BelPov Est','Under 18 BelPov MOE' ]
df1 = pd.read_csv(filename,usecols=fields)
df1.columns=["Geography", "est","moe"]
df1.columns

Index(['Geography', 'est', 'moe'], dtype='object')

In [42]:
mergedf1 = pd.merge(df_3,df1)
mergedf1

Geography  PctKidEst  PctKidMoe    zBflo2   est  \
0                  Abanda CDP, Alabama  24.712644  31.291041  0.072993     0   
1              Abbeville city, Alabama  20.688226   4.842623 -0.891406  31.1   
2             Adamsville city, Alabama  21.694451   5.021630 -0.531627  18.6   
3                Addison town, Alabama  19.583333   9.239504 -0.665180  51.8   
4                  Akron town, Alabama  18.238994   9.560045 -0.873986  55.2   
5         Alexander City city, Alabama  24.712245   2.138275  1.044431  45.7   
6              Alexandria CDP, Alabama  20.379839   5.000982 -0.964435   9.2   
7             Aliceville city, Alabama  21.953221   5.687021 -0.395239  61.7   
8                Allgood town, Alabama  21.237458  12.134203 -0.282670  42.5   
9                Altoona town, Alabama  18.719212   6.598968 -1.145156  19.1   
10             Andalusia city, Alabama  21.845398   2.019872 -1.174568  22.2   
11              Anderson town, Alabama  15.702479   8.748035 -1.431224  28.9   
12                  Arab city, Alabama  23.021583   2.035995 -0.238437  26.2   
13               Ardmore town, Alabama  20.939597   7.322344 -0.534634  29.5   
14                  Argo town, Alabama  21.570031   6.026889 -0.477373     0   
15                Ariton town, Alabama  23.702665   8.018948  0.077549  55.6   
16               Ashford town, Alabama  21.910367   3.547554 -0.650149  27.3   
17               Ashland town, Alabama  20.051151   7.506219 -0.715932  33.7   
18              Ashville city, Alabama  26.840753   6.532959  0.883359  24.4   
19                Atmore city, Alabama  23.179253   4.141127 -0.057161  33.5   
20               Attalla city, Alabama  24.952049   5.913156  0.451562  16.8   
21          Autaugaville town, Alabama  18.050941   8.458001 -1.024073  31.9   
22                  Avon town, Alabama  26.211454   7.418047  0.639097  11.8   
23                   Axis CDP, Alabama  34.056886  17.985516  0.981338  11.6   
24                Babbie town, Alabama  25.631769  10.294114  0.368416  54.2   
25             Baileyton town, Alabama  28.784404  13.737942  0.653473  33.1   
26             Bakerhill town, Alabama  15.744681  11.706505 -1.064241  18.9   
27               Ballplay CDP, Alabama  22.022741   7.873558 -0.271417     0   
28                 Banks town, Alabama  25.454545  23.088046  0.151768  57.1   
29            Bear Creek town, Alabama  19.071730   7.493730 -0.931722  40.3   
...                                ...        ...        ...       ...   ...   
18774          Riverside town, Wyoming   0.000000  27.906977 -1.374670     -   
18775           Riverton city, Wyoming  22.688284   2.054700 -0.496855  17.1   
18776         Rock River town, Wyoming  10.937500  13.094589 -1.555104     0   
18777      Rolling Hills town, Wyoming  27.906977  14.899854  0.505738     0   
18778           Saratoga town, Wyoming  22.743682   8.065234 -0.118178   3.2   
18779               Shell CDP, Wyoming  27.218935  30.785285  0.208100     0   
18780           Sheridan city, Wyoming  23.217828   1.280499 -0.128499   6.4   
18781           Shoshoni town, Wyoming  17.261905   9.620123 -1.035431    31   
18782           Sinclair town, Wyoming  22.926829  11.430810 -0.057114   2.1   
18783              Slater CDP, Wyoming  32.098765  30.016165  0.480834  46.2   
18784       Sleepy Hollow CDP, Wyoming  16.386555   9.348691 -1.219267     0   
18785               Smoot CDP, Wyoming  60.231660  51.338359  1.182559     0   
18786       South Greeley CDP, Wyoming  25.603757   4.968632  0.751601  21.9   
18787          South Park CDP, Wyoming  31.602280   8.939571  1.521326     0   
18788  Star Valley Ranch town, Wyoming  20.282069   8.412986 -0.593922   2.6   
18789               Story CDP, Wyoming  18.573046  15.572586 -0.501651     0   
18790           Sundance town, Wyoming  27.817404  10.342528  0.713986     0   
18791           Superior town, Wyoming  29.765886  11.047647  0.958379  12.4   
18792       Teton Village CDP

In [43]:
mergedf1['est']=mergedf1['est'].apply(lambda x: pd.to_numeric(x, errors='coerce'))
mergedf1['moe']=mergedf1['moe'].apply(lambda x: pd.to_numeric(x, errors='coerce'))

In [44]:
bufEst = mergedf1.loc[mergedf1.Geography=='Buffalo city, New York',['est']].values[0,0]
bufMoe = mergedf1.loc[mergedf1.Geography=='Buffalo city, New York',['moe']].values[0,0]
bufEst,bufMoe

(47.2, 2.4)

In [45]:
mergedf1['zBflo3']=(mergedf1['est']-bufEst)/((mergedf1['moe']/1.645)**2+(bufMoe/1.645)**2)**.5
mergedf1['zBflo3'][:10]

0   -1.782209
1   -1.657224
2   -3.137862
3    0.285452
4    0.473317
5   -0.251825
6   -5.321012
7    1.313515
8   -0.236522
9   -3.323440
Name: zBflo3, dtype: float64

In [46]:
mergedf1.loc[mergedf1.zBflo3 > 1.645,['Geography','est','moe']] # percent poverty greater than Buffalo, 90% confidence

Geography    est   moe
43                  Boligee town, Alabama   84.9  17.1
55                 Broomtown CDP, Alabama  100.0  52.0
58                Camp Hill town, Alabama   86.1  14.5
103                  Dayton town, Alabama   93.8  34.4
156             Garden City town, Alabama   74.6  25.7
168             Gordonville town, Alabama   77.8  26.8
189              Hayneville town, Alabama   75.0  15.7
197             Hobson City town, Alabama   76.5  15.8
219                Kellyton town, Alabama   91.2  17.1
244              Louisville town, Alabama   94.5   8.4
255                  Malcolm CDP, Alabama  100.0  14.2
280                  Mosses town, Alabama   74.1  12.1
296                 Newbern town, Alabama   82.1  24.3
322                  Perdido CDP, Alabama   78.8  18.6
361               Rock Mills CDP, Alabama   85.7  32.2
373                   Selma city, Alabama   56.1   7.7
440              White Hall town, Alabama   68.8  20.2
449                    York city, Alabama   70.0  18.2
450                     Adak city, Alaska   87.0  17.5
546          Northway Village CDP, Alaska   80.0  30.6
616                 Ash Fork CDP, Arizona   82.8  23.7
619           Bitter Springs CDP, Arizona   88.0  11.0
620               Blackwater CDP, Arizona   69.4  18.6
647            Dolan Springs CDP, Arizona   95.4   9.3
651                 Drysdale CDP, Arizona   79.1  31.0
673                   Gisela CDP, Arizona   69.7  20.3
684          Icehouse Canyon CDP, Arizona   88.0  35.1
693                 Klagetoh CDP, Arizona   76.1  23.9
727          Padre Ranchitos CDP, Arizona  100.0  23.8
752    Sacaton Flats Village CDP, Arizona   79.4  28.8
...                                   ...    ...   ...
17711             Markham CDP, Washington  100.0  29.4
17759            Packwood CDP, Washington  100.0  49.4
17810         South Bend city, Washington   67.1  18.0
17884      Amherstdale CDP, West Virginia   90.4  35.2
17901          Berwind CDP, West Virginia   78.1  28.9
17913        Bradshaw town, West Virginia   75.0  21.0
17915          Brenton CDP, West Virginia  100.0  52.0
17919              Bud CDP, West Virginia  100.0  37.0
17927     Cedar Grove town, West Virginia   82.0  20.2
17960       Elizabeth town, West Virginia   79.8  14.1
17974        Fort Gay town, West Virginia   68.2  17.5
18005        Henlawson CDP, West Virginia  100.0  39.9
18024          Kincaid CDP, West Virginia  100.0  49.6
18049         Maybeury CDP, West Virginia  100.0  46.8
18067     New Richmond CDP, West Virginia  100.0  44.0
18069       Northfork town, West Virginia   73.5  25.1
18077          Pageton CDP, West Virginia  100.0  42.0
18099             Rand CDP, West Virginia   72.5  19.7
18100           Raysal CDP, West Virginia  100.0  21.3
18108        Robinette CDP, West Virginia   75.4  23.6
18161      Williamson city, West Virginia   75.9  16.7
18225           Bowler village, Wisconsin   71.0  17.9
18306               Dunbar CDP, Wisconsin   91.3  19.4
18445      Little Sturgeon CDP, Wisconsin  100.0  34.9
18481            Mole Lake CDP, Wisconsin   63.4  15.2
18551         Radisson village, Wisconsin   93.5  12.0
18686                 Albin town, Wyoming   80.3  32.8
18727               Frannie town, Wyoming   77.4  25.7
18736              Hartrandt CDP, Wyoming   86.2  27.2
18763                 Osmond CDP, Wyoming  100.0  51.6

[416 rows x 3 columns]

In [47]:
mergedf1['zBflo3']=mergedf1['zBflo3'].abs()
mergedf1.loc[mergedf1.zBflo3 <= 1.645,['Geography','est','moe']] # percent poverty similar to Buffalo, 90% confidence

Geography    est    moe
3                Addison town, Alabama   51.8   26.4
4                  Akron town, Alabama   55.2   27.7
5         Alexander City city, Alabama   45.7    9.5
7             Aliceville city, Alabama   61.7   18.0
8                Allgood town, Alabama   42.5   32.6
11              Anderson town, Alabama   28.9   23.1
13               Ardmore town, Alabama   29.5   18.0
15                Ariton town, Alabama   55.6   20.0
17               Ashland town, Alabama   33.7   20.1
19                Atmore city, Alabama   33.5   13.5
21          Autaugaville town, Alabama   31.9   24.1
24                Babbie town, Alabama   54.2   26.0
25             Baileyton town, Alabama   33.1   21.6
26             Bakerhill town, Alabama   18.9   30.7
28                 Banks town, Alabama   57.1   49.6
29            Bear Creek town, Alabama   40.3   23.0
30              Beatrice town, Alabama   30.8   45.2
32               Belgreen CDP, Alabama    0.0  100.0
33                  Belk town, Alabama   36.0   29.1
34         Belle Fontaine CDP, Alabama   20.9   29.8
35                Benton town, Alabama    0.0  100.0
36                 Berry town, Alabama   48.6   23.1
37              Bessemer city, Alabama   47.7    7.0
38           Billingsley town, Alabama   30.0   41.8
40          Blountsville town, Alabama   53.2   14.5
45             Bon Secour CDP, Alabama   20.9   39.8
49            Bridgeport city, Alabama   51.9   18.9
50              Brighton city, Alabama   31.0   20.8
52           Bristow Cove CDP, Alabama   48.5   47.1
53             Brookside town, Alabama   41.0   25.1
...                                ...    ...    ...
18690   Arrowhead Springs CDP, Wyoming    0.0   68.0
18691              Arvada CDP, Wyoming    0.0  100.0
18693            Bairoil town, Wyoming    0.0   51.6
18706          Clearmont town, Wyoming   25.0   47.3
18707     Clearview Acres CDP, Wyoming   28.1   35.1
18713       Diamondville town, Wyoming   21.5   30.6
18714              Dixon town, Wyoming   52.6   50.4
18716   East Thermopolis town, Wyoming   30.8   35.1
18718           Edgerton town, Wyoming   34.4   42.7
18720         Encampment town, Wyoming   26.0   23.6
18724              Farson CDP, Wyoming    0.0   51.6
18725        Fort Bridger CDP, Wyoming    0.0   48.1
18726    Fox Farm-College CDP, Wyoming   31.2   20.9
18728             Freedom CDP, Wyoming   35.8   46.9
18731            Granger town, Wyoming    0.0   64.7
18734           Guernsey town, Wyoming   30.2   17.5
18742          James Town CDP, Wyoming   27.5   33.3
18745              Kirby town, Wyoming   41.7   43.6
18747          La Grange town, Wyoming   46.6   30.4
18756       Medicine Bow town, Wyoming   37.2   41.4
18759       Mountain View CDP, Wyoming    0.0   69.9
18762               Opal town, Wyoming    0.0   74.1
18768      Point of Rocks CDP, Wyoming    0.0   50.9
18776         Rock River town, Wyoming    0.0   79.3
18781           Shoshoni town, Wyoming   31.0   22.8
18783              Slater CDP, Wyoming   46.2   12.0
18796         Turnerville CDP, Wyoming    0.0   69.9
18798        Van Tassell town, Wyoming  100.0  100.0
18800          Wamsutter town, Wyoming    0.0   49.4
18803  Woods Landing-Jelm CDP, Wyoming    0.0  100.0

[6730 rows x 3 columns]

## Result
6730 places have percent poverty equal to Buffalo where population under 18 similar to Buffalo'.
416 place percent poverty of child greater than Buffalo